edb/pgsql/compiler/dml.py

```diff
@@ -382,6 +382,7 @@ def merge_iterator(
         put_iterator_bond(iterator, select)
         relctx.include_rvar(
             select, iterator_rvar,
+            aspects=('source', 'value', iterator.aspect),
```

Comment:
This was changed from `'value', ('source' if some condition)`, to `'value', 'source', 'iterator/identity`.

Are you sure that this `source` condition is not needed anymore?

Reply:
I'm pretty confident it isn't a bug right now, but actually it is kind of sloppy in the same way that that iterator sloppiness led to your trouble, so...